common/types/ClientOptions.ts

```diff
@@ -0,0 +1,29 @@
+import { Modify } from "./utils";
+import * as API from '../../ably';
+
+export default interface ClientOptions extends API.Types.ClientOptions {
+  restAgentOptions?: { keepAlive: boolean, maxSockets: number };
+  pushFullWait?: boolean;
+  checkChannelsOnResume?: boolean;
+  plugins?: Record<string, unknown>;
```

Comment:
- Since there's only one accepted plugin -- and it's very easy to type the wrong name -- why not have it typed, at least the key -- `plugins?: { vcdiff?: any }`
- Why is plugins in the 'non-public' part of the client options (in here instaed of in ably.d.ts)?

Reply:
Good point, I've added both the suggestions